Brokerages/Oanda/OandaBrokerage.cs

```diff
@@ -0,0 +1,1052 @@
+        /// <returns></returns>
+        public TradesResponse GetTradeList(Dictionary<string, string> requestParams = null)
+        {
+            var requestString = EndpointResolver.ResolveEndpoint(OandaEnvironment, Server.Account) + "accounts/" + AccountId + "/trades";
+            return MakeRequest<TradesResponse>(requestString, "GET", requestParams);
+        }
+        
+        /// <summary>
+        /// Modify the specified order, updating it with the parameters provided
+        /// </summary>
+        /// <param name="orderId">the identifier of the order to update</param>
+        /// <param name="requestParams">the parameters to update (name, value pairs)</param>
+        public void UpdateOrder(long orderId, Dictionary<string, string> requestParams)
+        {
+            var orderRequest = EndpointResolver.ResolveEndpoint(OandaEnvironment, Server.Account) + "accounts/" + AccountId + "/orders/" + orderId;
+
+            var order = MakeRequest<DataType.Order>(orderRequest);
+            if (order != null && order.id > 0)
+            {
+                var requestString = EndpointResolver.ResolveEndpoint(OandaEnvironment, Server.Account) + "accounts/" + AccountId + "/orders/" + orderId;
+                try
+                {
+                    MakeRequestWithBody<DataType.Order>(requestString, "PATCH", requestParams);
+                     
+                } 
+                catch (Exception)
+                {
+                }
+                
+            } 
+            else
+            {
+
+                OnMessage(new BrokerageMessageEvent(BrokerageMessageType.Warning, "UpdateFailed", "Failed to update Oanda order id: " + orderId + "."));
+            }
+        }
+
+        /// <summary>
+        /// Retrieves the details for a given order ID
+        /// </summary>
+        /// <param name="orderId">the id of the order to retrieve</param>
+        /// <returns>Order object containing the order details</returns>
+        public DataType.Order GetOrderDetails(int orderId)
+        {
+            var requestString = EndpointResolver.ResolveEndpoint(OandaEnvironment, Server.Account) + "accounts/" + AccountId + "/orders/" + orderId;
+            var order = MakeRequest<DataType.Order>(requestString);
+            return order;
+        }
+
+        /// <summary>
+        /// Retrieves the current rate for each of a list of instruments
+        /// </summary>
+        /// <param name="instruments">the list of instruments to check</param>
+        /// <returns>List of Price objects with the current price for each instrument</returns>
+        public List<Price> GetRates(List<Instrument> instruments)
+        {
+            var requestBuilder = new StringBuilder(EndpointResolver.ResolveEndpoint(OandaEnvironment, Server.Rates) + "prices?instruments=");
+
+            foreach (var instrument in instruments)
+            {
+                requestBuilder.Append(instrument.instrument + ",");
+            }
```

Comment:
hehe, yea, I tend to be minimalist in some cases, I think it stems from my desire to have each method to have focus on its primary goal, so a setup step like this as a one liner fits that desire, but toe-may-toe-; toe-mah-toe :D